Tasks/NuGetInstaller/nugetinstaller.ts

```diff
@@ -65,8 +66,9 @@ if (!tl.filePathSupplied('nuGetPath')) {
 var serviceUri = tl.getEndpointUrl("SYSTEMVSSCONNECTION", false);
 
 //find nuget location to use
-var nuGetPathToUse = ngToolRunner.locateTool('nuget.exe', userNuGetPath);
-var credProviderPath = ngToolRunner.locateTool('CredentialProvider.TeamBuild.exe', null, true);
+var nuGetExeVariants = ['nuget.exe', 'NuGet.exe', 'nuget', 'NuGet']
```

Comment:
On Windows, it is still case-insensitive. For non-windows, I only added the variants with caps because the original filename of NuGet.exe has caps, so that's a fairly likely packaging error when we release new versions.